constraint-solver/src/solver/base.rs

```diff
@@ -219,18 +219,57 @@ where
     /// If there is exactly one assignment that does not lead to a contradiction,
     /// apply it. This might be expensive.
     fn exhaustive_search(&mut self) -> Result<bool, Error> {
-        let assignments = exhaustive_search::get_unique_assignments(
-            self.constraint_system.system(),
-            &*self,
-            &self.bus_interaction_handler,
-        )?;
+        log::debug!("Starting exhaustive search...");
+        let mut variable_sets =
+            exhaustive_search::get_brute_force_candidates(self.constraint_system.system(), &*self)
+                .collect_vec();
+        // Start with small sets to make larger ones redundant after some assignments.
+        variable_sets.sort_by_key(|set| set.len());
+
+        log::debug!(
+            "Found {} sets of variables with few possible assignments. Checking each set...",
+            variable_sets.len()
+        );
 
         let mut progress = false;
-        for (variable, value) in &assignments {
-            progress |=
-                self.apply_range_constraint_update(variable, RangeConstraint::from_value(*value));
-        }
 
+        let mut error_sets = BTreeSet::new();
```

Comment:
Is error set the right name? If I understand this right, it just contains the variables that don't have an assignment, which isn't really an error? `unsuccessfully_processed_sets`? ^^ Not quite as catchy, but describes it better IMO.